src/pybind/mgr/cephadm/services/smb.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from typing import Any, Dict, Iterator, List, Tuple, cast
+
+from ceph.deployment.service_spec import ServiceSpec, SMBSpec
+
+from orchestrator import DaemonDescription
+from .cephadmservice import (
+    AuthEntity,
+    CephService,
+    CephadmDaemonDeploySpec,
+    simplified_keyring,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SMBService(CephService):
+    TYPE = 'smb'
+
+    def config(self, spec: ServiceSpec) -> None:
+        assert self.TYPE == spec.service_type
+        logger.warning('config is a no-op')
```

Comment:
This is the only one I meant to leave at the warning level, because I couldn't figure out when this method was supposed to get called.
The others are just me being lazy and I forgot to turn them into debugs. Will fix.